### DISCUSSION TITLE
case expression with array as OutpuType dont work

### DISCUSSION THREAD

**Discussion by kauly** [maplibre/maplibre-gl-js#2770] [Q&A]
I have the following expression and it's not working:

```js
{
  paint: {
    'icon-translate': ['case', ['==', ['feature-state', 'hover'], true], ['literal', [5, 5]], ['literal', [0, 0]]]
  },
}
```
I'm using the "literal" because it's a [recommended workaround ](https://github.com/mapbox/mapbox-gl-js/issues/11442), but it's not working. There is nothing about this in the docs.

**Comment by HarelM** [comment]
Is the problem with the types definition or doe this not work in runtime?
If this doesn't work in runtime then it means that you either wrote something wrong or that you are using data driven definition for propertied that are not defined as data driven.
I know this doesn't help much, but ChatGPT might be able to help in your case maybe.

**Reply by kauly** [reply]
Thank you for your answer, it's a runtime error but the expression is correct and icon-translate is a data-driven property. I decide to do another thing on hover.
